enhancements/machine-config/install-time-support-for-image-mode-on-openshift.md

```diff
@@ -0,0 +1,342 @@
+
+#### Story 2: Regulated Environment Deployment
+
+As a cluster admin in a regulated environment, I want to include security agents and compliance tools in my OS image during installation, so that my cluster meets compliance requirements from the moment it comes online, without requiring post-installation configuration.
+
+#### Story 3: Edge Computing Deployment
+
+As a cluster admin deploying edge computing solutions, I want to include edge-specific drivers and monitoring agents in my OS image during installation, so that my edge nodes are immediately functional without requiring network connectivity for post-installation customization.
+
+### Goals
+
+- Enable cluster admins to use Image Mode on OpenShift at install time
+- Support custom OS image builds for master, worker, and any other custom pools during bootstrap
+- Maintain OpenShift's single-click upgrade experience maintaining customizations across upgrades
+- Ensure control plane and worker nodes can all use custom OS images
+- Provide a declarative approach to OS customization through MachineOSConfig CRD
+
+### Non-Goals
+
+- Replacing the existing ignition-based configuration system entirely
+- Supporting arbitrary OS distributions beyond RHCOS-based images
+- Providing external CI/CD integration for image builds (builds occur on-cluster)
+
+## Proposal
+
+### Workflow Description
+
+The install time Image Mode on OpenShift workflow integrates with the existing OpenShift installation process in the following steps.
+
+#### Step 1: Installation Configuration
+
+1. Cluster admin runs the installer (openshift-install) to generate install time manifests
+2. Cluster admin specifies "on-cluster" image mode enablement in `install-config.yaml`
+3. Cluster admin provides `MachineOSConfig` (MOSC) yamls for master and worker MachineConfigPools
+4. Cluster admin provides/creates secret(s) needed to push and pull built images to and from the external registry
+5. Installer runs the bootkube service to combine all the manifests in the install directory to create a massive ignition file
+
+#### Step 2: Bootstrap Phase
+
+1. Bootstrap Machine Config Operator(MCO) parses the massive ignition file created by the bootkube service during the installer phase
+2. Bootstrap MCO/MCC parses the ignition file to generate the necessary MachineConfigs
+3. Bootstrap MCO/MCC pods establishes connection with external image registry (for this enhancement, running a registry on the bootstrap node is a non-goal)
+4. Once the rendered-MC is generated, bootstrap MCO/MCC starts the "on-cluster" image build process by creating a pod to build the custom OS image for the master MCP and push the built image to the external registry
+5. Bootstrap MCO/MCC creates a MachineConfig to override the `osImageURL` to point to the newly built image for the master pool
+6. Bootstrap node creates the `control plane` nodes (master pool) with the stock AMI
```

Comment:
AMI would imply AWS and I think we should make the workflow platform agnostic, so let's drop that reference